searx/webapp.py

```diff
@@ -247,37 +254,7 @@ def _get_browser_or_settings_language(req, lang_list):
         locale = match_language(lang, lang_list, fallback=None)
         if locale is not None:
             return locale
-    return settings['search']['default_lang'] or 'en'
-
-
-@babel.localeselector
-def get_locale():
-    if 'locale' in request.form\
```

Comment:
See pre_request function:
https://github.com/searxng/searxng/blob/a925254c3d19c6a3badf43469816fc1b497e2990/searx/webapp.py#L530-L542

If `locale in request.form` then the value has been parsed, and `request.preferences.get_value('locale')` contains the same value.